sql/item_strfunc.cc

```diff
@@ -1449,6 +1449,42 @@ String *Item_func_sformat::val_str(String *res)
   return null_value ? NULL : res;
 }
 
+
+#include <openssl/rand.h>
+
+bool Item_func_random_bytes::fix_length_and_dec() { return FALSE; }
+
+String *Item_func_random_bytes::val_str(String *str)
+{
+  longlong count= args[0]->val_int();
+
+  if (args[0]->null_value)
+    goto err;
+  null_value= 0;
+
+  if (count < 1 || count > 1024)
+  {
+    char buf[256];
+    String msg(buf, sizeof(buf), system_charset_info);
+    msg.length(0);
+    print(&msg, QT_NO_DATA_EXPANSION);
+    my_error(ER_DATA_OUT_OF_RANGE, MYF(0), "length", msg.c_ptr_safe());
+    return make_empty_result(str);
+  }
+
+  if (str->alloc((uint) count))
+    goto err;
+
+  str->length(count);
+  RAND_bytes((unsigned char *) str->ptr(), count);
```

Comment:
Check the return value of RAND_bytes. Any reason not to use RAND_priv_bytes?

Reply:
>[RAND_priv_bytes()] is intended to be used for generating values that should remain private. 
I doubt it is the case for a user-level function

Reply:
Given the current use of `RAND_bytes` (wrapped by `my_random_bytes`, largely around storage encryption), using `RAND_priv_bytes` here instead of `RAND_bytes` will keep the server and user-level randomness separated. I haven't looked at the full cryptographic risks here, however if we think there are risks of some quantification, and if `RAND_priv_bytes` isn't significantly worse performance, perhaps we should just it. 

Please do handle the return case of `RAND_{priv_,}bytes` failing.